source/base/ReferenceCounter.ooc

```diff
@@ -1,20 +1,29 @@
 import threading/Thread
 import Synchronized
 
-ReferenceCounter: class extends Synchronized {
+ReferenceCounter: class {
 	_target: Object
 	count: static Int = 0
 	_count: Int = 0
 	_kill := false
+	_lock := Mutex new(MutexType Safe)
+	isSafe: Bool {
+		get { !this _lock instanceOf?(MutexUnsafe) }
+		set(value) {
+			if (this isSafe != value) {
+				this _lock free()
+				this _lock = Mutex new(value ? MutexType Safe : MutexType Unsafe)
```

Comment:
Maybe this is a dumb question (still brewing my coffee), but, would anyone ever want to use the global mutex here?